linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -0,0 +1,964 @@
+        checkTree(expr, env)
+        env
+
+      // JavaScript expressions
+
+      case JSNew(ctor, args) =>
+        checkTree(ctor, env)
+        checkTreeOrSpreads(args, env)
+        env
+
+      case JSPrivateSelect(qualifier, className, field) =>
+        checkTree(qualifier, env)
+        env
+
+      case JSSelect(qualifier, item) =>
+        checkTree(qualifier, env)
+        checkTree(item, env)
+        env
+
+      case JSFunctionApply(fun, args) =>
+        checkTree(fun, env)
+        checkTreeOrSpreads(args, env)
+        env
+
+      case JSMethodApply(receiver, method, args) =>
+        checkTree(receiver, env)
+        checkTree(method, env)
+        checkTreeOrSpreads(args, env)
+        env
+
+      case JSSuperSelect(superClass, qualifier, item) =>
+        checkTree(superClass, env)
+        checkTree(qualifier, env)
+        checkTree(item, env)
+        env
+
+      case JSSuperMethodCall(superClass, receiver, method, args) =>
+        checkTree(superClass, env)
+        checkTree(receiver, env)
+        checkTree(method, env)
+        checkTreeOrSpreads(args, env)
+        env
+
+      case JSSuperConstructorCall(args) =>
+        reportError("illegal JSSuperConstructorCall")
+        checkTreeOrSpreads(args, env)
+        env
+
+      case JSImportCall(arg) =>
+        checkTree(arg, env)
+        env
+
+      case JSNewTarget() =>
+        if (!env.hasNewTarget)
+          reportError("Cannot refer to `new.target` outside of a JS class constructor or non-arrow function")
+        env
+
+      case JSImportMeta() =>
+        env
+
+      case LoadJSConstructor(_) =>
+        env
+
+      case LoadJSModule(_) =>
+        env
+
+      case JSUnaryOp(_, lhs) =>
+        checkTree(lhs, env)
+        env
+
+      case JSBinaryOp(_, lhs, rhs) =>
+        checkTree(lhs, env)
+        checkTree(rhs, env)
+        env
+
+      case JSArrayConstr(items) =>
+        checkTreeOrSpreads(items, env)
+        env
+
+      case JSObjectConstr(fields) =>
+        for ((key, value) <- fields) {
+          checkTree(key, env)
+          checkTree(value, env)
+        }
+        env
+
+      case JSGlobalRef(_) =>
+        env
+
+      case JSTypeOfGlobalRef(_) =>
+        env
+
+      case JSLinkingInfo() =>
+        env
+
+      // Literals
+
+      case ClassOf(typeRef) =>
+        typeRef match {
+          case NullRef | NothingRef =>
+            reportError(i"Invalid classOf[$typeRef]")
+          case typeRef: ArrayTypeRef =>
+            checkArrayTypeRef(typeRef)
+          case _ =>
+            // ok
+        }
+        env
+
+      case _: Literal =>
+        env
+
+      // Atomic expressions
+
+      case VarRef(LocalIdent(name)) =>
+        env.locals.get(name).fold[Unit] {
+          reportError(i"Cannot find variable $name in scope")
+        } { localDef =>
+          if (tree.tpe != localDef.tpe)
+            reportError(i"Variable $name of type ${localDef.tpe} "+
+                i"typed as ${tree.tpe}")
```

Comment:
Probably fine to write on one line:
```suggestion
          if (tree.tpe != localDef.tpe)
            reportError(i"Variable $name of type ${localDef.tpe} typed as ${tree.tpe}")
```